src/components/form/form-file-input.jsx

```diff
@@ -31,6 +31,12 @@ class FormFileInput extends React.Component {
     this.state = { values: props.values };
   }
 
+  componentWillReceiveProps(nextProps) {
```

Comment:
Can you please explain why this is needed? It seems this should automatically be how it works

Reply:
The `FormFileInput` component is a stateful react component. It receives props and assigns them to state once during component initialization:

```
constructor(props) {
    super(props);
    this.state = { values: props.values };
  }
```

When the component receives new props, the state is not being updated automatically and later in code when we read `this.state.value`, it still contains the old value. `componentWillReceiveProps` lifecycle method has access to the next props values. Here we can update the state using props and applying our custom logic. We can specify use cases when we want to update the store and when not.

FYI `componentWillReceiveProps` along with list of other lifecycle methods will be deprecated after version 17: https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops

But currently, we use `componentWillReceiveProps` in all our plugins, therefore I don't refactor it in current PR. And, to be honest, I am not sure how exactly new lifecycle methods work :) So it needs to be investigated separately.

Reply:
Ok! Does it work if you comment out this method? Wondering because there doesn’t seem to be any custom logic here (unless I’m totally missing something?), and react should do the “===“ check as default right?

Reply:
No, it doesn't :) this is why I put it here. This is a ticket: https://jira.mongodb.org/browse/COMPASS-3711?filter=-1

And I check for `nextProps.values !== this.state.values` to reduce amount of renders. Because the component is being rerendered each time the state is changed and the state contains only `value` property.